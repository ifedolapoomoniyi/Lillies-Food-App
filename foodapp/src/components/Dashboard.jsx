import React, { useState } from "react";
import "../styles/Dashboard.css";
import ImageComp from "./ImageComp";
import { Food } from "../data/data";

const Dashboard = () => {

	return (
		<div className="Dashboard">
			<div className="dashboard-header">
				<h3>Good morning, {localStorage.getItem("name")}!</h3>
				<img src="" alt="Profile" />
			</div>
			<div className="dashboard-body">
					{Food.map((item, key) => {
						return (
				<div className="dashboard-image">
							<ImageComp
								name={item.name}
								text={item.description}
								image={require("../imgs/" + item.img + ".png")}
                                price={item.price}
                                cart={item.cart}
                                about={item.about}
							>
                            </ImageComp>
				</div>
                
						);
					})}
			</div>
		</div>
	);
};

export default Dashboard;
