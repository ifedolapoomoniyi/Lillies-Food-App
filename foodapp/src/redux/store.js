import { configureStore } from '@reduxjs/toolkit';
import cartReducer from './cartSlice';

const store = configureStore({
    reducer: {
        theme: cartReducer,
    },
});

export default store;