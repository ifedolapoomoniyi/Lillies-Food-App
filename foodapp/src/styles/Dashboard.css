.Dashboard{
    width: 100%;
    padding: 3rem;
}

.dashboard-header{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 2rem;
}

.dashboard-header img{
    height: 50px;
    border-radius: 50%;
}

.dashboard-header-welcome{
    display: flex;
    flex-direction: column;
}

.dashboard-body{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 2rem;
    justify-content: space-evenly;

}