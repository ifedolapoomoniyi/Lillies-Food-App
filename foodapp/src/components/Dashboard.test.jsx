import React from "react"; 
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Dashboard from "./Dashboard";
import { Provider } from "react-redux";
import store from "../redux/store";

test("renders component correctly", () => {
    const view = render(
    <Provider store={store}>
        <Dashboard/>
    </Provider>
    )
    expect(view).toMatchSnapshot();
})

test("renders text correctly", () => {
    render(
    <Provider store={store}>
        <Dashboard/>
    </Provider>
    )
    const testMessage = 'Good morning';
    expect(screen.getByText(testMessage)).toBeInTheDocument();
})

// test("renders text correctly", () => {
//     expect(screen.getByText("Good morning")).toBeInTheDocument();
//     expect(screen.getByText("delicious meal")).toBeInTheDocument();
// })