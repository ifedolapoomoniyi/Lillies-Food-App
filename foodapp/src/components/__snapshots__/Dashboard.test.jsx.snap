// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Dashboard"
      >
        <div
          class="dashboard-header"
        >
          <div
            class="dashboard-header-welcome"
          >
            <h3>
              Good morning, 
              !
            </h3>
            <div>
              What delicious meal are you craving today?
            </div>
          </div>
          <img
            alt="Profile"
            src="Avatar.jpg"
          />
        </div>
        <div
          class="dashboard-body"
        >
          <div
            class="dashboard-image"
          >
            <div
              class="ImageComp"
            >
              <img
                alt=""
                src="img4.png"
              />
              <h3>
                Burger
              </h3>
              <p>
                A hot burger made from the rarest of meats
              </p>
              <div
                class="price-and-cart"
              >
                <div
                  class="price"
                >
                  #
                  1000
                </div>
                <div
                  class="add-to-cart"
                >
                  <button>
                    Add to cart
                  </button>
                </div>
              </div>
              <div
                class="Toastify"
              />
            </div>
          </div>
          <div
            class="dashboard-image"
          >
            <div
              class="ImageComp"
            >
              <img
                alt=""
                src="img2.png"
              />
              <h3>
                StirFry Spaghetti
              </h3>
              <p>
                In House Spaghetti made by our best chef
              </p>
              <div
                class="price-and-cart"
              >
                <div
                  class="price"
                >
                  #
                  3000
                </div>
                <div
                  class="add-to-cart"
                >
                  <button>
                    Add to cart
                  </button>
                </div>
              </div>
              <div
                class="Toastify"
              />
            </div>
          </div>
          <div
            class="dashboard-image"
          >
            <div
              class="ImageComp"
            >
              <img
                alt=""
                src="img3.png"
              />
              <h3>
                MeatBalls
              </h3>
              <p>
                Who doesn't love meat with some balls
              </p>
              <div
                class="price-and-cart"
              >
                <div
                  class="price"
                >
                  #
                  5000
                </div>
                <div
                  class="add-to-cart"
                >
                  <button>
                    Add to cart
                  </button>
                </div>
              </div>
              <div
                class="Toastify"
              />
            </div>
          </div>
          <div
            class="dashboard-image"
          >
            <div
              class="ImageComp"
            >
              <img
                alt=""
                src="img1.png"
              />
              <h3>
                Daily special
              </h3>
              <p>
                The day's special meal, in a delicious package
              </p>
              <div
                class="price-and-cart"
              >
                <div
                  class="price"
                >
                  #
                  4000
                </div>
                <div
                  class="add-to-cart"
                >
                  <button>
                    Add to cart
                  </button>
                </div>
              </div>
              <div
                class="Toastify"
              />
            </div>
          </div>
          <div
            class="dashboard-image"
          >
            <div
              class="ImageComp"
            >
              <img
                alt=""
                src="img2.png"
              />
              <h3>
                Banana Delicacy
              </h3>
              <p>
                Bananas wrapped in royal goodness
              </p>
              <div
                class="price-and-cart"
              >
                <div
                  class="price"
                >
                  #
                  1500
                </div>
                <div
                  class="add-to-cart"
                >
                  <button>
                    Add to cart
                  </button>
                </div>
              </div>
              <div
                class="Toastify"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Dashboard"
    >
      <div
        class="dashboard-header"
      >
        <div
          class="dashboard-header-welcome"
        >
          <h3>
            Good morning, 
            !
          </h3>
          <div>
            What delicious meal are you craving today?
          </div>
        </div>
        <img
          alt="Profile"
          src="Avatar.jpg"
        />
      </div>
      <div
        class="dashboard-body"
      >
        <div
          class="dashboard-image"
        >
          <div
            class="ImageComp"
          >
            <img
              alt=""
              src="img4.png"
            />
            <h3>
              Burger
            </h3>
            <p>
              A hot burger made from the rarest of meats
            </p>
            <div
              class="price-and-cart"
            >
              <div
                class="price"
              >
                #
                1000
              </div>
              <div
                class="add-to-cart"
              >
                <button>
                  Add to cart
                </button>
              </div>
            </div>
            <div
              class="Toastify"
            />
          </div>
        </div>
        <div
          class="dashboard-image"
        >
          <div
            class="ImageComp"
          >
            <img
              alt=""
              src="img2.png"
            />
            <h3>
              StirFry Spaghetti
            </h3>
            <p>
              In House Spaghetti made by our best chef
            </p>
            <div
              class="price-and-cart"
            >
              <div
                class="price"
              >
                #
                3000
              </div>
              <div
                class="add-to-cart"
              >
                <button>
                  Add to cart
                </button>
              </div>
            </div>
            <div
              class="Toastify"
            />
          </div>
        </div>
        <div
          class="dashboard-image"
        >
          <div
            class="ImageComp"
          >
            <img
              alt=""
              src="img3.png"
            />
            <h3>
              MeatBalls
            </h3>
            <p>
              Who doesn't love meat with some balls
            </p>
            <div
              class="price-and-cart"
            >
              <div
                class="price"
              >
                #
                5000
              </div>
              <div
                class="add-to-cart"
              >
                <button>
                  Add to cart
                </button>
              </div>
            </div>
            <div
              class="Toastify"
            />
          </div>
        </div>
        <div
          class="dashboard-image"
        >
          <div
            class="ImageComp"
          >
            <img
              alt=""
              src="img1.png"
            />
            <h3>
              Daily special
            </h3>
            <p>
              The day's special meal, in a delicious package
            </p>
            <div
              class="price-and-cart"
            >
              <div
                class="price"
              >
                #
                4000
              </div>
              <div
                class="add-to-cart"
              >
                <button>
                  Add to cart
                </button>
              </div>
            </div>
            <div
              class="Toastify"
            />
          </div>
        </div>
        <div
          class="dashboard-image"
        >
          <div
            class="ImageComp"
          >
            <img
              alt=""
              src="img2.png"
            />
            <h3>
              Banana Delicacy
            </h3>
            <p>
              Bananas wrapped in royal goodness
            </p>
            <div
              class="price-and-cart"
            >
              <div
                class="price"
              >
                #
                1500
              </div>
              <div
                class="add-to-cart"
              >
                <button>
                  Add to cart
                </button>
              </div>
            </div>
            <div
              class="Toastify"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
